contracts/nexus-gateway/src/nexus.rs

```diff
@@ -68,24 +68,37 @@ impl From<connection_router_api::Message> for Message {
     }
 }
 
-impl From<Message> for connection_router_api::Message {
-    fn from(msg: Message) -> Self {
-        Self {
+impl TryFrom<Message> for connection_router_api::Message {
+    type Error = Report<ContractError>;
+    fn try_from(msg: Message) -> Result<Self, ContractError> {
+        // this string is just used in error messages
+        let msg_id = format!(
+            "{}-{}",
+            msg.source_tx_id.as_ref().encode_hex::<String>(),
+            msg.source_tx_index
+        );
+        Ok(Self {
             cc_id: CrossChainId {
                 chain: msg.source_chain,
-                id: format!(
-                    "{}:{}",
-                    msg.source_tx_id.as_ref().encode_hex::<String>(),
-                    msg.source_tx_index
-                )
+                id: HexTxHashAndEventIndex {
+                    tx_hash: <[u8; 32]>::try_from(msg.source_tx_id.as_ref().as_slice())
+                        .map_err(|_| axelar_wasm_std::msg_id::Error::InvalidTxHash(msg_id.clone()))
+                        .change_context(ContractError::InvalidMessageId(msg_id.clone()))?,
+                    event_index: u32::try_from(msg.source_tx_index)
+                        .map_err(|_| {
+                            axelar_wasm_std::msg_id::Error::EventIndexOverflow(msg_id.clone())
+                        })
+                        .change_context(ContractError::InvalidMessageId(msg_id))?,
+                }
+                .to_string()
                 .try_into()
-                .expect("cannot be empty"),
+                .expect("must serialize message id to non-empty string"),
```

Comment:
seems weird that this is a panic. If the underlying string was a constant I would agree, but here I'd prefer just to return an error if this fails (even if it never should, but it relies on multiple other parts to implicitly work as expected). You don't save anything by panicking here, there return type is a `Result` either way